MNT: Make error message clearer for n_neighbors (#23317)
scikit-learn/scikit-learn: doc/whats_new/v1.2.rst

  :class:`neighbors.RadiusNeighborsTransformer`.
  :pr:`24075` by :user:`Valentin Laurent <Valentin-Laurent>`.

> - |Fix| Improve error message for :class:`neighbors.LocalOutlierFactor`

Reviewer: now 1.2.1 would make more sense ;)

  :class:`neighbors.RadiusNeighborsTransformer`.
  :pr:`24075` by :user:`Valentin Laurent <Valentin-Laurent>`.

> - |Fix| Make error message clearer for `n_neighbors` when
>   :class:`neighbors.LocalOutlierFactor` is invoked when
>   `n_samples = n_neighbors`.
>   :pr:`23317` by :user:`Bharat Raghunathan <Bharat123rox>`.

Reviewer 2: What do you think of the following rewording:
```suggestion
- |Fix| Improve error message for :class:`neighbors.LocalOutlierFactor`
  when it is invoked with `n_samples = n_neighbors`.
  :pr:`23317` by :user:`Bharat Raghunathan <Bharat123rox>`.
```
